### ISSUE TITLE
IDE0059 - Unneccessary assignment is not reported in a Test class using XUnit and Mock

### ISSUE THREAD

**Issue by JuanZamudioGBM** [dotnet/roslyn#38885]
**Version Used**: 
16.3.0
.net core SDK 2.2.402

**Steps to Reproduce**:
I have some tests using Xunit and Mock, when  I use Setup in a mock IDE0059 is not reported.
Here is a little example

```csharp
using System;
using System.Threading.Tasks;
using Gbm.Api.Crm.Repositories.Abstractions;
using Moq;
using Xunit;


public class Ide0059Test
{
    [Fact]
    public void Test_IDE0059_Not_Firing()
    {
        Guid contractId = Guid.NewGuid();
        Guid accountId = Guid.NewGuid(); // ide0059 not reported

        var contractRepository = new Mock<IContractRepository>();
        contractRepository
           .Setup(m => m.FindByIdAsync(It.Is<Guid>(i => i == contractId)))
           .Returns(Task.FromResult((Crm.Repositories.Abstractions.DataContracts.Contract)null))
           .Verifiable();
    }
    [Fact]
    public void Test_IDE0059_Firing()
    {
        Guid contractId = Guid.NewGuid();
        Guid accountId = Guid.NewGuid(); // ide0059 error
        if (contractId != default)
        {
            Console.WriteLine(contractId);
        }
    }
}
```

**Expected Behavior**:
IDE0059 warning
**Actual Behavior**:
No warning

**Comment by mavasani** [comment]
Likely a case we are bailing out due to delegates and captures being involved

**Comment by mavasani** [comment]
Yep, validated that is the case. Analysis is just conservative here.
